File: app/src/main/java/ai/elimu/appstore/util/SharedPreferencesHelper.java
package ai.elimu.appstore.util;

import android.content.Context;
import android.content.SharedPreferences;

import timber.log.Timber;

public class SharedPreferencesHelper {

    private static final String SHARED_PREFS = "shared_prefs";

    public static final String PREF_APP_VERSION_CODE = "pref_app_version_code";

    public static void clearAllPreferences(Context context) {
        Timber.w("clearAllPreferences");
        SharedPreferences sharedPreferences = context.getSharedPreferences(SHARED_PREFS, Context.MODE_PRIVATE);
        sharedPreferences.edit().clear().apply();
    }

    public static void storeAppVersionCode(Context context, int appVersionCode) {
        Timber.i("storeAppVersionCode");
        SharedPreferences sharedPreferences = context.getSharedPreferences(SHARED_PREFS, Context.MODE_PRIVATE);
        sharedPreferences.edit().putInt(PREF_APP_VERSION_CODE, appVersionCode).apply();
    }

    public static int getAppVersionCode(Context context) {
        Timber.i("getAppVersionCode");
        SharedPreferences sharedPreferences = context.getSharedPreferences(SHARED_PREFS, Context.MODE_PRIVATE);
        return sharedPreferences.getInt(PREF_APP_VERSION_CODE, 0);
    }
}

File: app/src/main/java/ai/elimu/appstore/util/SharedPreferencesHelper.kt
package ai.elimu.appstore.util

import ai.elimu.model.v2.enums.Language
import android.content.Context
import android.text.TextUtils
import timber.log.Timber
import androidx.core.content.edit

object SharedPreferencesHelper {
    private const val SHARED_PREFS = "shared_prefs"

    private const val PREF_APP_VERSION_CODE: String = "pref_app_version_code"
    private const val PREF_LANGUAGE: String = "pref_language"

    @JvmStatic
    fun storeAppVersionCode(context: Context, appVersionCode: Int) {
        Timber.i("storeAppVersionCode")
        val sharedPreferences = context.getSharedPreferences(SHARED_PREFS, Context.MODE_PRIVATE)
        sharedPreferences.edit { putInt(PREF_APP_VERSION_CODE, appVersionCode) }
    }

    @JvmStatic
    fun getAppVersionCode(context: Context): Int {
        Timber.i("getAppVersionCode")
        val sharedPreferences = context.getSharedPreferences(SHARED_PREFS, Context.MODE_PRIVATE)
        return sharedPreferences.getInt(PREF_APP_VERSION_CODE, 0)
    }


    @JvmStatic
    fun storeLanguage(context: Context, language: Language) {
        Timber.i("storeLanguage")
        val sharedPreferences = context.getSharedPreferences(SHARED_PREFS, Context.MODE_PRIVATE)
        sharedPreferences.edit { putString(PREF_LANGUAGE, language.toString()) }
    }

    @JvmStatic
    fun getLanguage(context: Context): Language? {
        Timber.i("getLanguage")
        val sharedPreferences = context.getSharedPreferences(SHARED_PREFS, Context.MODE_PRIVATE)
        val languageAsString = sharedPreferences.getString(PREF_LANGUAGE, null)
        if (TextUtils.isEmpty(languageAsString)) {
            return null
        } else {
            val language = Language.valueOf(
                languageAsString!!
            )
            return language
        }
    }
}
